x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRulesTests.java

```diff
@@ -1377,10 +1382,131 @@ public void testExactMatchRLike() throws Exception {
+        FieldAttribute fa = getFieldAttribute();
+        Literal one = of(1);
+        Literal two = of(2);
+
+        Or or = new Or(EMPTY, equalsOf(fa, one), new In(EMPTY, fa, asList(one, two)));
+        Expression e = new CombineDisjunctionsToIn().rule(or);
+        assertEquals(In.class, e.getClass());
+        In in = (In) e;
+        assertEquals(fa, in.value());
+        assertThat(in.list(), contains(one, two));
+    }
+
+    public void testSingleValueInToEquals() throws Exception {
+        FieldAttribute fa = getFieldAttribute();
+        Literal one = of(1);
+
+        Equals equals = equalsOf(fa, one);
+        Or or = new Or(EMPTY, equals, new In(EMPTY, fa, singletonList(one)));
+        Expression e = new CombineDisjunctionsToIn().rule(or);
+        assertEquals(equals, e);
+    }
+
+    public void testEqualsBehindAnd() throws Exception {
+        FieldAttribute fa = getFieldAttribute();
+        Literal one = of(1);
+        Literal two = of(2);
+
+        And and = new And(EMPTY, equalsOf(fa, one), equalsOf(fa, two));
+        Expression e = new CombineDisjunctionsToIn().rule(and);
+        assertEquals(and, e);
+    }
+
+    public void testTwoEqualsDifferentFields() throws Exception {
+        FieldAttribute fieldOne = getFieldAttribute("one");
+        FieldAttribute fieldTwo = getFieldAttribute("two");
+        Literal one = of(1);
+        Literal two = of(2);
+
+        Or or = new Or(EMPTY, equalsOf(fieldOne, one), equalsOf(fieldTwo, two));
+        Expression e = new CombineDisjunctionsToIn().rule(or);
+        assertEquals(e, e);
+    }
+
+    public void testMultipleIn() throws Exception {
+        FieldAttribute fa = getFieldAttribute();
+        Literal one = of(1);
+        Literal two = of(2);
+        Literal three = of(3);
+
+        Or firstOr = new Or(EMPTY, new In(EMPTY, fa, singletonList(one)), new In(EMPTY, fa, singletonList(two)));
+        Or secondOr = new Or(EMPTY, firstOr, new In(EMPTY, fa, singletonList(three)));
+        Expression e = new CombineDisjunctionsToIn().rule(secondOr);
+        assertEquals(In.class, e.getClass());
+        In in = (In) e;
+        assertEquals(fa, in.value());
+        assertThat(in.list(), contains(one, two, three));
+    }
+
+    public void testOrWithNonCombinableExpressions() throws Exception {
+        FieldAttribute fa = getFieldAttribute();
+        Literal one = of(1);
+        Literal two = of(2);
+        Literal three = of(3);
+
+        Or firstOr = new Or(EMPTY, new In(EMPTY, fa, singletonList(one)), lessThanOf(fa, two));
+        Or secondOr = new Or(EMPTY, firstOr, new In(EMPTY, fa, singletonList(three)));
+        Expression e = new CombineDisjunctionsToIn().rule(secondOr);
+        assertEquals(Or.class, e.getClass());
+        Or or = (Or) e;
+        assertEquals(LessThan.class, or.right().getClass());
+        LessThan lt = (LessThan) or.right();
+        assertEquals(fa, lt.left());
+        assertEquals(two, lt.right());
```

Comment:
Super nit-picky. Would just test that there are no changes:

```suggestion
        assertEquals(or.right(), firstOr.right());
```